plugins/catalog-backend-module-github/api-report.md

```diff
@@ -157,4 +180,67 @@ export class GithubOrgReaderProcessor implements CatalogProcessor {
     emit: CatalogProcessorEmit,
   ): Promise<boolean>;
 }
+
+// @public
+export type GitHubTeam = {
+  slug: string;
+  combinedSlug: string;
+  name?: string;
+  description?: string;
+  avatarUrl?: string;
+  editTeamUrl?: string;
+  parentTeam?: GitHubTeam;
+  members: Connection<GitHubUser>;
```

Comment:
Im not sure this is possible or expected.

In the `defaultOrganizationTeamTransformer` we use the connection to iterate through the users in the GH team. It allows us to prevent extra API calls if we know we already have all the users in the team.

```
if (!team.members.pageInfo.hasNextPage) {
    // We got all the members in one go, run the fast path
    for (const user of team.members.nodes) {
      memberNames.push(user.login);
    }
  } else {
    // There were more than a hundred immediate members - run the slow
    // path of fetching them explicitly
    const { members } = await getTeamMembers(ctx.client, ctx.org, team.slug);
    for (const userLogin of members) {
      memberNames.push(userLogin);
    }
  }
```

The other less explicit way of doing this would be to check if there are 99 or less people in the returned query. We would then try and paginate through the team members.

Reply:
For our scale this extra API call is not great as we have over 1k teams and most have less than 100 members and could be gathered in single api calls. but changing this to always call to check is going to add a significant amount of api calls and makes us hit our rate limit a lot easier. 

We already have lots of issues with rate limits as we have over 7k repos and the discovery mechanism is sub optimal at the moment (might be my next PR)

Reply:
I think you misunderstand. Maybe this type is used for more than one purpose, that might explain the confusion.

What I want is that whatever _the transformer functions consume_ should not have a Connection object on them. Hence, the public interface should not contain a Connection object. At the point in time that the transformer is called, _the framework code_ that's outside of the transformer function should already have done what's necessary to read the users. _That code_ can indeed make use of connections and graphql and whatever it wants. But _the transform function_ should be blind to that fact.

Reply:
Indeed in the vast majority of cases, _the framework code_ should not need to make any additional calls to construct the `GithubUser[]` array that is then sent to the transformer

Reply:
Ah ok so you want me to refactor so that the framework code materialises the list of users per group and send that to the transformer. My aim was to try and keep the existing transformers as close as possible to the existing implementation.

I will see how well that change can be made without breaking things.